src/ec/packages/ec-component-button/button.story.js

```diff
@@ -37,30 +31,59 @@ uiIcons.forEach(icon => {
 // labels for the groups.
 const requiredGroupId = 'Mandatory elements';
 const optionalGroupId = 'Optional elements';
+const statusGroupId = 'States';
 
 // Preserve the adapted specs.
 const prepareButton = data => {
+  data.disabled = boolean('disabled', data.disabled, statusGroupId);
   data.label = text('label', data.label, requiredGroupId);
 
   data.variant = select(
     'variant (default value: primary)',
     [data.variant],
     data.variant,
-    optionalGroupId
+    requiredGroupId
   );
 
-  data.disabled = boolean('disabled', data.disabled, optionalGroupId);
+  data.extra_classes = text('extra_classes', '', optionalGroupId);
 
-  data.extra_classes = text(
-    'extra_classes (comma separated)',
-    '',
-    optionalGroupId
-  );
-  data.extra_attributes = object(
-    'extra_attributes',
-    { name: '', value: '' },
-    optionalGroupId
-  );
+  const attribute1Name = text('extra_attributes[0].name', '', optionalGroupId);
+  // First attribute.
+  if (attribute1Name !== '') {
+    data.extra_attributes = [];
+    let attribute = {};
+    const attribute1Value = text(
+      'extra_attributes[0].value',
+      '',
+      optionalGroupId
+    );
+    const attribute2Name = text(
+      'extra_attributes[1].name',
+      '',
+      optionalGroupId
+    );
+    attribute.name = attribute1Name;
+    if (attribute1Value !== '') {
+      attribute.value = attribute1Value;
+    }
+    data.extra_attributes.push(attribute);
+    // Second attribute.
+    if (attribute2Name !== '') {
```

Comment:
Is it really necessary to be able to set a second attribute? 
My feeling is that we are having more and more code. As extra_classes and extra_attributes is part of every-components, we can maybe have these knobs more globally?

Reply:
It's true, we need code to implement functionalities.. :)
I though about having these knobs, which are always the same from a utility sort of, but i don't know, that could be tricky, actually, and most likely misbehaving, we need to define these knobs in the stories, but we can give it a try. 
The reason why i don't feel disturbed by the "code" needed to implement these "optional knobs"for extra_classes and extra_attribute is that it's always the same and if we decide to go for a "utility" for instance, this is a standard change to be applied to all the components with no particular effort.
I'll give it a try and see what consequencies this might have, for the "second" attribute yes, i think is important to show the fact that this parameter accepts an array and you can set multiple extra_attributes.

Reply:
ok @planctus are you creating a ticket to investigate utilities,
if yes I can merge this and we will see if this can be improved globally?

Reply:
ah well, no i did it here and it seems to work.. :)  
if we like the solution we merge it and i adapt all the other pull requests opened, i think we will add also other functions to this node module.. ;)
To make it work you need to  run yarn